web/modules/custom/joinup_federation/src/Plugin/pipeline/Step/BrokenReferences.php

```diff
@@ -0,0 +1,177 @@
+
+  /**
+   * The entity field manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * Creates a new pipeline step plugin instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param array $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\rdf_entity\Database\Driver\sparql\Connection $sparql
+   *   The SPARQL database connection.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager
+   *   The entity field manager service.
+   * @param \Drupal\rdf_schema_field_validation\SchemaFieldValidatorInterface $rdf_schema_field_validator
+   *   The RDF schema field validator service.
+   */
+  public function __construct(array $configuration, $plugin_id, array $plugin_definition, Connection $sparql, EntityTypeManagerInterface $entity_type_manager, EntityFieldManagerInterface $entity_field_manager, SchemaFieldValidatorInterface $rdf_schema_field_validator) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $sparql);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityFieldManager = $entity_field_manager;
+    $this->rdfSchemaFieldValidator = $rdf_schema_field_validator;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('sparql_endpoint'),
+      $container->get('entity_type.manager'),
+      $container->get('entity_field.manager'),
+      $container->get('rdf_schema_field_validation.schema_field_validator')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute() {
+    $blacklist = array_flip($this->getPersistentDataValue('blacklist'));
+    $ids = array_keys($this->getPersistentDataValue('entities'));
+    /** @var \Drupal\rdf_entity\RdfInterface $entity */
+    foreach ($this->getRdfStorage()->loadMultiple($ids, ['staging']) as $id => $entity) {
+      $changed = FALSE;
+      $reference_fields = $this->getAdmsSchemaEntityReferenceFields($entity->bundle(), ['rdf_entity', 'taxonomy_term']);
+
+      foreach ($reference_fields as $field_name => $target_entity_type_id) {
+        /** @var \Drupal\Core\Field\EntityReferenceFieldItemListInterface $field */
+        $field = $entity->get($field_name);
+
+        // Remove references to entities that were blacklisted by the user.
+        if ($blacklist && ($target_entity_type_id === 'rdf_entity')) {
+          $changed |= $this->removeBlacklistedReferences($field, $blacklist);
```

Comment:
I actually had to look this up. Is there a reason we are asking for an integer to be returned here? In any case, the `$changed` variable is initialized as `FALSE`. Either initialize it as `0` or use plain `=` signs in this line.

Reply:
We don't have the same operator for booleans, something that should do `$changed = $changed || ....`.